progressist/__init__.py

```diff
@@ -133,6 +133,9 @@ def throttled(self):
                self._last_render + self.throttle.seconds > time.time()):
                 return True
             self._last_render = time.time()
+        else:
+            raise ValueError('Invalid type for throttle: '
+                             '{}'.format(type(self.throttle)))
```

Comment:
I would check in the constructor instead.

Let's say you're doing an operation that has the potential to be destructive, and it should either all be done or not at all. (say, a package manager transaction :wink: )

Raising the `ValueError` in the middle of the operation (when printing the progress bar) is a pretty hostile way to tell the programmer they passed an invalid value for the `throttle` argument. :smile: